score.py

```diff
@@ -128,6 +134,95 @@ def run_inference(images_with_header: List[ImageWithHeader],
     return segmentation
 
 
+def extract_zipped_dicom_series(zip_file_path: Path, extraction_folder: Path) -> None:
+    """
+    Unzip a zip file. Assume that it is a single DICOM series, and extract all
+    DICOM files into the extraction folder.
+
+    :param zip_file_path: Path to zip file.
+    :param extraction_folder: Path to extraction folder.
+    """
+    with zipfile.ZipFile(zip_file_path, 'r') as zip_file:
+        for zipped_file in zip_file.infolist():
+            # try to extract only DICOM files, but these need not have
+            # an extension
+            if not zipped_file.is_dir() and (
+                    zipped_file.filename.endswith('.dcm') or '.' not in zipped_file.filename):
+                # discard the path, if any, to just get the filename and suffix
+                zipped_file.filename = os.path.basename(zipped_file.filename)
+                zip_file.extract(zipped_file, str(extraction_folder))
+
+
+def convert_zipped_dicom_to_nifti(test_image: Path, model_folder: Path) -> Tuple[Path, Path]:
+    """
+    Given a zip file, extract DICOM series and convert to Nifti format.
+
+    :param test_image: Path to a zip file.
```

Comment:
Can you add a bit more documentation here about what this function creates? It has to many sideeffects to be left unmentioned. and change variable names? test_image -> zipped_dicom_folder for example. Why is model_folder called model_folder, if it's just scratch?